### PR TITLE
Coll han update file reading 5.0.x

### PR DESCRIPTION
**Pull request by FlorentGermain-Bull**
Rebase of #10828 + #10456 + #10458 + #10983 + #10992  on v5.0.x branch

### CODE REVIEW

**Comment by awlauria** [comment]
ok to test

File: ompi/mca/coll/han/coll_han_component.c [hunk 1]
```diff
@@ -160,22 +170,37 @@ static bool is_simple_implemented(COLLTYPE_T coll)
...
+    { INTER_NODE, "inter_node" },
+    { GLOBAL_COMMUNICATOR, "global_communicator" },
+    { 0 }
+};
+
 /*
  * Stringifier for topological level
  */
-const char* mca_coll_han_topo_lvl_to_str(TOPO_LVL_T topo_lvl)
+const char* mca_coll_han_topo_lvl_to_str(TOPO_LVL_T topo_lvl_id)
```
Comment (devreal) [inline ompi/mca/coll/han/coll_han_component.c:187]: I realize t hat this breaks the common naming scheme. It should be `ompi_coll_han_topo_lvl_to_str`. Given that this has been merged into main already this (and other occurrences) should be fixed in a separate PR.

**Review by devreal — CHANGES_REQUESTED** [review]
Can you please include cherry-picks of https://github.com/open-mpi/ompi/pull/10983 and https://github.com/open-mpi/ompi/pull/10992 here since they are fixing issues in these changes?

**Review by bosilca — APPROVED** [review]

**Comment by FlorentGermain-Bull** [comment]
> Can you please include cherry-picks of #10983 and #10992 here since they are fixing issues in these changes?

done
+ rebase on v5.0.x

**Comment by jsquyres** [comment]
bot:ibm:retest

**Review by devreal — APPROVED** [review]

**Pull request merged**